packages/bff/src/services/agreementService.ts

```diff
@@ -93,6 +97,145 @@ export function agreementServiceBuilder(clients: PagoPAInteropBeClients) {
 
       return enhanceAgreement(agreement, clients, ctx);
     },
+
+    async getAgreementsEserviceProducers({
+      ctx,
+      offset,
+      limit,
+      requesterId,
+      states,
+      eServiceName,
+    }: {
+      offset: number;
+      limit: number;
+      requesterId: string;
+      states: agreementApi.AgreementState[];
+      ctx: WithLogger<BffAppContext>;
+      eServiceName?: string;
+    }): Promise<bffApi.CompactEServicesLight> {
+      ctx.logger.info(
+        `Retrieving producer eservices from agreement filtered by eservice name ${eServiceName}, offset ${offset}, limit ${limit}`
+      );
+
+      const eservices = await agreementProcessClient.getAgreementEServices({
+        queries: {
+          offset,
+          limit,
+          eServiceName,
+          producersIds: [requesterId],
+          states,
+        },
+        headers: ctx.headers,
+      });
+
+      if (eServiceName && eServiceName.length < 3) {
+        return emptyPagination(offset, limit);
+      }
+
+      return {
+        pagination: {
+          limit,
+          offset,
+          totalCount: eservices.totalCount,
+        },
+        results: eservices.results.map((e) => toCompactEserviceLight(e)),
+      };
+    },
+
+    async getAgreementsEserviceConsumers(
+      offset: number,
+      limit: number,
+      requesterId: string,
+      ctx: WithLogger<BffAppContext>,
+      eServiceName?: string
+    ) {
+      ctx.logger.info(
+        `Retrieving consumer eservices from agreement filtered by eservice name ${eServiceName}, offset ${offset}, limit ${limit}`
+      );
+
+      const eservices = await agreementProcessClient.getAgreementEServices({
+        queries: {
+          offset,
+          limit,
+          eServiceName,
+          consumersIds: [requesterId],
+        },
+        headers: ctx.headers,
+      });
+
+      if (eServiceName && eServiceName.length < 3) {
+        return emptyPagination(offset, limit);
+      }
```

Comment:
Could this check be placed before the `getAgreementEServices` call to avoid unnecessary HTTP request?

Reply:
Yes of course, i didn't notice it and i copied the Scala code, thanks for the heads up